test/common.py

```diff
@@ -2040,7 +2041,7 @@ def get_poppler_library(self, env_init=None):
         os.path.join('third_party', 'poppler'),
         [os.path.join('utils', 'pdftoppm.o'), os.path.join('utils', 'parseargs.o'), os.path.join('poppler', '.libs', 'libpoppler.a')],
         env_init=env_init,
-        configure_args=['--disable-libjpeg', '--disable-libpng', '--disable-poppler-qt', '--disable-poppler-qt4', '--disable-cms', '--disable-cairo-output', '--disable-abiword-output', '--disable-shared'])
```

Comment:
`--build` is the build machine architecture... that isn't `wasm32-emscripten` is it?

Reply:
No, but `--target` doesn't work, and I'm not sure `--host` does either.

Reply:
I'm not sure why we don't need this elsewhere though?  If its needed should it no be part of emconfigure maybe?  How have we not needed this before?

Reply:
`--host` doesn't work either. The problem is that config.sub doesn't recognize arm64-darwin, not that it doesn't recognize wasm. So we need to override the build system; and if we have to do that and lie about the build system anyway, maybe it makes sense to set it to wasm? The theoretical alternative would be to set `--build` to whatever we need to get it to work at all, and then set `--target` to wasm (although that only works for poppler and not for freetype.

Reply:
Oh I see.. maybe we should update the config.sub then?

Reply:
IIRC config.sub is packged along with configure and the other outputs of autoconf. So if you are using a source distribution (i.e. not running autoconf/automake yourself) then it may not work to just update that part. If there is a new source package created with a newer version of autoconf/automake then it might work. (of course presumably such a distribution would also recognize arem64-darwin as a build system).

Reply:
I'm pretty sure its normal/expected to update just `config.sub` and `config.guess` before building source packages.  I remember doing that when I was building debian packages.

I suggest we just update the copies that we have checked in to the latest release.

Reply:
```
./test/third_party/poppler/config.sub
./test/third_party/freealut/admin/autotools/config.sub
./test/third_party/enet/config.sub
./test/third_party/freetype/builds/unix/config.sub
./test/third_party/bullet/config.sub
```

Reply:
I updated config.sub/config.guess for poppler and freetype. It fixed the problem that the arm64 build machine was unrecognized, but it still seems that Poppler is passing mac linker flags unless the host is manually overridden. My guess is that either config.sub isn't guessing the host/target correctly based on emcc output, or (more likely) poppler's configure is incorrectly setting link flags based on the host rather than the target.

Reply:
But it looks like its passing on the mac arm64 bot, no?

Reply:
Oh I see you put the `--host` back in.

So this test has simply never run correctly on mac before?

Maybe lets just disabled the warning on the command line instead?  Using `--host=wasm32-emscripten` just seems strange.

Reply:
Oh wait.. sorry `--host` is not actually strange.  I was confusing it with `--build`..

Reply:
Yeah I'm guessing this has never run correctly on mac, but I don't know for sure. I think --host is the same as --build when you're not building a compiler. So it is strange to use --host=wasm32-emscripten. But since the script isn't really right for cross-compiling, I think faking a native wasm build is no more strange than trying to work around issues caused by doing the build as mac-native.

Reply:
Actually no, [apparently](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html) host is what we want here and maybe it's target that's not relevant for non-compilers.

Reply:
`--host` and `--build` being the same means you are not cross compiling.   `--host` and `--build` being different means you are cross compiling.

`--target` is the thing that only matters when you are building a compiler.

emscripten is a little bit odd because even though we are cross compiling we don't actually tell autoconf that.  i.e. we pretend that emcc is not actually a cross compiler.. and this is kind of true since it can actually run the generated `.js` files locally on the build system.

However, in this case telling audoconf that we are actually cross compiling doesn't seem unreasonable.

Reply:
i.e. this is how `emconfigure` works on my local machine normally:
```
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... /usr/local/google/home/sbc/dev/wasm/emscripten/emcc
```

i.e. it does not detect that its cross compiling.